### DISCUSSION TITLE
Seeking software technology assistance

### DISCUSSION THREAD

**Discussion by Curitis-Mr** [jankae/LibreVNA#271] [Q&A]
I am a PCBA DIY worker, currently converting libre VNA to a 4-layer PCB. The power supply DCDC has been changed to 5-12V 3A, and now there are two issues. How to download the underlying driver for FPAGA XC6SLX9-2TQG144C. STM32G431CBU6 project compilation error, unable to download; If that friend can solve the problem, please contact their email“ curitis.sale@gmail.com ”As a reward, you will receive a free set of Liber VNA. The gift is only available within China, and overseas express delivery may be more complicated, but you can also handle the postage yourself. Please, my friends.

**Comment by jankae** [comment]
Hi, I am unsure of what exactly you need assistance with but I'll try to answer any questions I can spot:

> How to download the underlying driver for FPAGA XC6SLX9-2TQG144C

There is no driver required for the FPGA. Did you maybe mean the bitstream? You *can* manually load that over JTAG but it is not required. See also [this](https://github.com/jankae/LibreVNA/blob/master/Documentation/DeveloperInfo/BuildAndFlash.md#getting-a-new-board-running) link.

> STM32G431CBU6 project compilation error

It would help if you could post the actual error message. Also, you can search old issues as you are not the first one to run into errors there.

> unable to download

No idea what that is referring to. Compiled binaries for the embedded firmware as well as the GUI are available on Github. There are no other things required or available for download.

**Reply by Curitis-Mr** [reply]
Thanks for the reply. Now that ISE has been discontinued and the current computer system is win11, which version of Vivado should be installed to open the project and compile it?

**Reply by jankae** [reply]
None. As far as I know, Vivado does not support the Spartan-6. You need to use ISE if you want to synthesize the bitstream yourself.
What is your goal? If you just want to use the firmware as is, there is no need to compile/synthesize anything yourself.

**Reply by Curitis-Mr** [reply]
My current FPAGA XC6SLX9-2TQG144C is an empty chip. I want to download the FPGA bitstream. My method is to use JTAG. Of course, there are documents telling me that I can directly update it using the GUI. This way I can also achieve my goal. Target.
